src/metabase/api/collection.clj

```diff
@@ -1063,33 +1063,77 @@
-     :count (count dashboard-question-candidates)}))
+  (let [cards-in-collection (t2/hydrate (t2/select :model/Card {:where [:= :collection_id collection-id]}) :in_dashboards)]
+    (filter card/sole-dashboard-id cards-in-collection)))
+
+(mu/defn- present-dashboard-question-candidate
+  [{:keys [in_dashboards] :as card}]
+  (-> card
+      (select-keys [:id :name :description])
+      (assoc :sole_dashboard_info (-> in_dashboards first (select-keys [:id :name :description])))))
+
+(mu/defn- present-dashboard-question-candidates
+  [cards]
+  {:data (map present-dashboard-question-candidate cards)
+   :count (count cards)})
 
 (api.macros/defendpoint :get "/:id/dashboard-question-candidates" :- ::DashboardQuestionCandidatesResponse
   "Find cards in this collection that can be moved into dashboards in this collection.
 
   To be eligible, a card must only appear in one dashboard (which is also in this collection), and must not already be a dashboard question."
-  [{:keys [id]} :- [:map [:id ms/PositiveInt]]
-   {:keys [sort_direction]} :- [:map [:sort_direction {:default :asc} [:enum :asc :desc]]]]
+  [{:keys [id]} :- [:map [:id ms/PositiveInt]]]
   (api/read-check :model/Collection id)
-  (dashboard-question-candidates id sort_direction))
+  (present-dashboard-question-candidates
+   (dashboard-question-candidates id)))
 
 (api.macros/defendpoint :get "/root/dashboard-question-candidates" :- ::DashboardQuestionCandidatesResponse
   "Find cards in the root collection that can be moved into dashboards in the root collection. (Same as the above endpoint, but for the root collection)"
+  []
+  (present-dashboard-question-candidates
+   (dashboard-question-candidates nil)))
+
+(mr/def ::MoveDashboardQuestionCandidatesResponse
+  [:map
+   [:success boolean?]
+   [:moved [:sequential ms/PositiveInt]]])
+
+(defn- move-dashboard-question-candidates
+  "Move dash"
+  [id card-ids]
+  (let [cards (cond->> (dashboard-question-candidates id)
+                (some? card-ids) (filter #(contains? card-ids (:id %))))]
+    (t2/with-transaction [_conn]
+      (doall
+       (for [{:as card :keys [in_dashboards]} cards]
+         (do
+           (card/update-card! {:card-before-update card
+                               :card-updates {:dashboard_id (-> in_dashboards first :id)}
+                               :actor @api/*current-user*
+                               :delete-old-dashcards? false})
+           (:id card)))))))
+
+(api.macros/defendpoint :post "/:id/move-dashboard-question-candidates" :- ::MoveDashboardQuestionCandidatesResponse
+  "Move candidate cards to the dashboards they appear in."
+  [{:keys [id]} :- [:map [:id ms/PositiveInt]]
+   _query-params
+   {:keys [card_ids]} :- [:maybe
+                          [:map [:card_ids {:optional true}
+                                 [:set ms/PositiveInt]]]]]
+  (api/read-check :model/Collection id)
+  {:success true
+   :moved (move-dashboard-question-candidates id card_ids)})
```

Comment:
Do we need an explicit hard-coded `:success` key in the response? Presumably if we return a list of moved cards then the operation succeeded